extensions/common/iam/identity-trust/identity-trust-sts-core/src/main/java/org/eclipse/edc/iam/identitytrust/sts/core/defaults/service/StsClientTokenGeneratorServiceImpl.java

```diff
@@ -0,0 +1,73 @@
+/*
+ *  Copyright (c) 2023 Bayerische Motoren Werke Aktiengesellschaft (BMW AG)
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Bayerische Motoren Werke Aktiengesellschaft (BMW AG) - initial API and implementation
+ *
+ */
+
+package org.eclipse.edc.iam.identitytrust.sts.core.defaults.service;
+
+import org.eclipse.edc.iam.identitytrust.sts.embedded.EmbeddedSecureTokenService;
+import org.eclipse.edc.iam.identitytrust.sts.model.StsClient;
+import org.eclipse.edc.iam.identitytrust.sts.model.StsClientTokenAdditionalParams;
+import org.eclipse.edc.iam.identitytrust.sts.service.StsClientTokenGeneratorService;
+import org.eclipse.edc.iam.identitytrust.sts.service.StsTokenGenerationProvider;
+import org.eclipse.edc.service.spi.result.ServiceResult;
+import org.eclipse.edc.spi.iam.TokenRepresentation;
+
+import java.time.Clock;
+import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+import static org.eclipse.edc.jwt.spi.JwtRegisteredClaimNames.AUDIENCE;
+import static org.eclipse.edc.jwt.spi.JwtRegisteredClaimNames.ISSUER;
+import static org.eclipse.edc.jwt.spi.JwtRegisteredClaimNames.SUBJECT;
+
+public class StsClientTokenGeneratorServiceImpl implements StsClientTokenGeneratorService {
+
+    private final Map<String, EmbeddedSecureTokenService> tokenGenerations;
+    // TODO configurable?
+    private final Integer capacity = 100;
```

Comment:
Primitive? Also, I think the caching strategy should be documented and configurable, e.g. the number

Reply:
Another question: how long can the token exist if it is not reaped due to excess capacity? For example is it possible that a token is cached beyond its expiration and inadvertently returned to the client?

Reply:
Here it's not cached the token but the token generator. It's just a premature optimization, we could also create  the `EmbeddedSecureTokenService`  every time a client ask for a token

Reply:
Cached removed :)